### PR TITLE
[PM-3779] idor allow the attacker to delete the victim domain

### PR DESCRIPTION
**Pull request by r-tome**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Fix a reported defect re: deleting organization domains.

## Code changes

* **src/Api/Controllers/OrganizationDomainController.cs:** Simplified the process by receiving `string` parameters as `Guid` from the beginning, instead of converting them to `Guid` later
* **src/Core/OrganizationFeatures/OrganizationDomains/DeleteOrganizationDomainCommand.cs:** Updated the method to directly receive the object, simplifying the command's functionality to focus solely on its intended action
* **src/Core/OrganizationFeatures/OrganizationDomains/GetOrganizationDomainByIdAndOrganizationIdQuery.cs:** Added the parameter `OrganizationId` to obtain a specific `OrganizationDomain`
* **src/Core/OrganizationFeatures/OrganizationDomains/GetOrganizationDomainByOrganizationIdQuery.cs:** Added "Async" to the end of the method name to comply with naming convention
* **src/Core/OrganizationFeatures/OrganizationDomains/Interfaces/IDeleteOrganizationDomainCommand.cs:** Updated the method to directly receive the object, simplifying the command's functionality to focus solely on its intended action
* **src/Core/OrganizationFeatures/OrganizationDomains/Interfaces/IGetOrganizationDomainByIdQuery.cs:** Added the parameter `OrganizationId` to obtain a specific `OrganizationDomain`
* **src/Core/OrganizationFeatures/OrganizationDomains/Interfaces/IGetOrganizationDomainByOrganizationIdQuery.cs:** Added "Async" to the end of the method name to comply with naming convention
* **src/Core/OrganizationFeatures/OrganizationDomains/Interfaces/IVerifyOrganizationDomainCommand.cs:** Updated the method to directly receive the object, simplifying the command's functionality to focus solely on its intended action
* **src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:** Updated the method to directly receive the object, simplifying the command's functionality to focus solely on its intended action
* **src/Core/OrganizationFeatures/OrganizationServiceCollectionExtensions.cs:** Updated the command name
* **src/Core/Repositories/IOrganizationDomainRepository.cs:** Added method to obtain an `OrganizationDomain` by both `Id` and `OrganizationId`
* **src/Infrastructure.Dapper/Repositories/OrganizationDomainRepository.cs:** Added method to obtain an `OrganizationDomain` by both `Id` and `OrganizationId`
* **src/Infrastructure.EntityFramework/Repositories/OrganizationDomainRepository.cs:** Added method to obtain an `OrganizationDomain` by both `Id` and `OrganizationId`
* **src/Sql/dbo/Stored Procedures/OrganizationDomain_ReadByIdAndOrganizationId.sql:** Stored procedure to obtain an `OrganizationDomain` by both `Id` and `OrganizationId`
* **test/Api.Test/Controllers/OrganizationDomainControllerTests.cs:** Updated unit tests
* **test/Core.Test/OrganizationFeatures/OrganizationDomains/DeleteOrganizationDomainCommandTests.cs:** Updated unit tests
* **test/Core.Test/OrganizationFeatures/OrganizationDomains/GetOrganizationDomainByIdAndOrganizationIdQueryTests.cs:** Query unit tests
* **test/Core.Test/OrganizationFeatures/OrganizationDomains/GetOrganizationDomainByIdQueryTests.cs:** Deleted old command unit tests
* **test/Core.Test/OrganizationFeatures/OrganizationDomains/GetOrganizationDomainByOrganizationIdQueryTests.cs:** Updated method name
* **test/Core.Test/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommandTests.cs:** Updated unit tests
* **util/Migrator/DbScripts/2023-09-29_00_OrgDomainReadByIdAndOrgId.sql:** SQL migration script to create stored procedure

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Infrastructure.EntityFramework/Repositories/OrganizationDomainRepository.cs [hunk 1]
```diff
@@ -93,6 +93,18 @@ public async Task<OrganizationDomainSsoDetailsData> GetOrganizationDomainSsoDeta
         return ssoDetails;
     }
 
+    public async Task<Core.Entities.OrganizationDomain> GetDomainByIdOrganizationIdAsync(Guid id, Guid orgId)
+    {
+        using var scope = ServiceScopeFactory.CreateScope();
+        var dbContext = GetDatabaseContext(scope);
+        var domain = await dbContext.OrganizationDomains
+            .Where(x => x.Id == id && x.OrganizationId == orgId)
+            .AsNoTracking()
```
Comment (vincentsalucci) [inline src/Infrastructure.EntityFramework/Repositories/OrganizationDomainRepository.cs:102]: Had to do some searching to figure out what this does! TIL

**Comment by vincentsalucci** [comment]
@r-tome Reviewed all of your changes but see there's a test failing. Once resolved, will green light!

File: test/Core.Test/OrganizationFeatures/OrganizationDomains/GetOrganizationDomainByIdAndOrganizationIdQueryTests.cs
Comment (vincentsalucci) [inline test/Core.Test/OrganizationFeatures/OrganizationDomains/GetOrganizationDomainByIdAndOrganizationIdQueryTests.cs:1]: Pretty lightweight test here 😄 : have any ideas/plans to add some more?

Comment (r-tome) [reply test/Core.Test/OrganizationFeatures/OrganizationDomains/GetOrganizationDomainByIdAndOrganizationIdQueryTests.cs:1]: Good shout, I have added a few more tests

**Review by vincentsalucci — APPROVED** [review]

**Pull request merged**